File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<string, int> hm;

        for(char ch : s){
            string key(1, ch);
            if(hm.find(key) == hm.end()){
                hm[key] = 0;
            }
            hm[key]++;
        }

        for(int i=0; i < s.length(); i++){
            string key(1, s[i]);
            if(hm[key] == 1){
                return i;
            }
        }

        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        hm = defaultdict(int)

        for c in s:
            hm[c] += 1

        for i in range(len(s)):
            if hm[s[i]] == 1:
                return i
        
        return -1